libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1781,6 +1781,23 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 			for (auto const& member: membersRemovedForStructConstructor)
 				msg += " " + member;
 		}
+		else if (
+			functionType->kind() == FunctionType::Kind::BareCall ||
+			functionType->kind() == FunctionType::Kind::BareCallCode ||
+			functionType->kind() == FunctionType::Kind::BareDelegateCall
+		)
+		{
+			if (arguments.empty())
+				msg += " This function requires a single bytes argument. Use \"\" as argument to provide empty calldata.";
+			else
+				msg += " This function requires a single bytes argument. If all your arguments are value types, you can use abi.encode(...) to properly generate it.";
+		}
+		else if (
+			functionType->kind() == FunctionType::Kind::SHA3 ||
+			functionType->kind() == FunctionType::Kind::SHA256 ||
+			functionType->kind() == FunctionType::Kind::RIPEMD160
+		)
+			msg += " This function requires a single bytes argument. Use abi.encodePacked(...) to properly encode the values.";
```

Comment:
I think this may be useful to be explained better: use `encodePacked` if the old behaviour is desired or `encode` if ABI encoded data is wished. Perhaps with a link to documentation.

Reply:
not sure if links to the documentation are useful in error messages, I think looking up the function names should be easy enough.